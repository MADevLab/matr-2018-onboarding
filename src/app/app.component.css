@import '~@angular/material/prebuilt-themes/indigo-pink.css';

.brand {
  margin-left: 10px;
  text-transform: uppercase;
}
