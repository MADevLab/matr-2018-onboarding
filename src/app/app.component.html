<mat-toolbar color="primary">
  <mat-icon svgIcon="map"></mat-icon>
  <span class="brand">Map Widget</span>
</mat-toolbar>
<div style="display: flex;">
  <div style="flex: 1 1 auto;">
    <ng-container *ngIf="dataSets$ | async as dataSets; else loadingDataSets">
      <br/>
      <app-data-set-options [dataSets]="dataSets"
                            (changeSet)="changeDataSet($event)"
                            [chosenSet]="chosenDataSet$ | async"
                            (changeDate)="changeViewDate($event)"></app-data-set-options>
      <div>[Add Regions]</div>
    </ng-container>
  </div>
</div>
<br/>
<div style="display: flex" *ngIf="dataSets$ | async as dataSets;">
  <div style="flex: 1 1 auto; min-width: 50%; min-height: 600px;">
    <mat-card>
      <ng-container *ngIf="mapData; else makeSelections">
        <app-map-view [title]="title"
                      [concept]="concept"
                      [dataPoints]="mapData"
                      [selectedRegions]="selectedRegions"></app-map-view>
      </ng-container>
    </mat-card>
  </div>
</div>
<ng-template #loadingDataSets>
  <mat-progress-bar mode="indeterminate"></mat-progress-bar>
</ng-template>
<ng-template #makeSelections>
  Please choose a date set and date to view the map
</ng-template>
