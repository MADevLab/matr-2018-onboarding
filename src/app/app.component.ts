import { Component, OnInit } from '@angular/core';
import { BehaviorSubject, timer } from 'rxjs';
import { filter, switchMap, switchMapTo, tap } from 'rxjs/operators';
import { DataSeriesService } from './data-series.service';
import { DataPoints } from './data-set.model';
import { DomSanitizer } from '@angular/platform-browser';
import { MatIconRegistry } from '@angular/material';
import { Regions } from './regions.model';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit {
  public dataSets$;

  public chosenDataSet$;

  public regions$;

  public concept: string = null;

  public title: string = null;

  public mapData: DataPoints = null;

  public selectedRegions: Regions[];


  private chosenDataSetId = new BehaviorSubject<number>(-1);

  /**
   * Sets of datapoints mapped to the date integers
   *
   * @memberof AppComponent
   */
  private dataPoints = new Map<number, DataPoints>();

  constructor(
    private dataSeriesService: DataSeriesService,
    iconRegistry: MatIconRegistry,
    sanitizer: DomSanitizer
  ) {
    iconRegistry.addSvgIcon(
      'map',
      sanitizer.bypassSecurityTrustResourceUrl('/assets/img/map.svg'));
  }

  public ngOnInit() {
    this.regions$ = this.dataSeriesService.getRegions();

    // uncomment once data-series-.service is working
    //this.dataSets$ = timer(3000).pipe(switchMapTo(this.dataSeriesService.getDataSets()));
    this.chosenDataSet$ = this.chosenDataSetId.pipe(
      filter(id => id !== -1),
      switchMap(id => this.dataSeriesService.getDataSet(id)),
      tap(dataSet => {
        this.dataPoints.clear();
        this.concept = dataSet.concept;
        this.title = dataSet.description;
        dataSet.data.forEach(dataPoint => {
          this.dataPoints.set(dataPoint.period, dataPoint);
        });
      })
    );
  }

  public changeDataSet(dataSetId) {
    this.chosenDataSetId.next(dataSetId);
  }

  public changeViewDate(date: number) {
    throw new Error('Missing Logic');
  }

  public changeRegions(values: Regions[]) {
    throw new Error('Missing Logic');
  }
}
