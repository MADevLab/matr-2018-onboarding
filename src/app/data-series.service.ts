import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { of, Observable } from 'rxjs';
import { DataSetOption } from './data-set-option.model';
import { DataSet } from './data-set.model';
import { Regions } from './regions.model';

@Injectable({
  providedIn: 'root'
})
export class DataSeriesService {
  public s3Url = 'https://s3-us-west-2.amazonaws.com/ma-cs-matr-2018-onboarding';

  // fallback if s3 is inaccessible. if requests are failing,
  // uncomment this line, and comment out line 12
  //public s3Url = 'assets';

  constructor(private http: HttpClient) {}

  public getDataSets() {
    throw new Error('Missing Logic');
  }

  public getDataSet(dataSetId: number): Observable<DataSet> {
    throw new Error('Missing Logic');

    // uncomment once missing logic is implemented
    // const url = `${this.s3Url}/${dataSets.get(dataSetId).dataUrl}`;

  }

  public getRegions(): Observable<Regions[]> {
    const url = 'assets/regions.json';

    throw new Error('Missing Logic');
  }

  private createDataSets() {
    const dataSets = new Map<number, DataSetOption>();
    dataSets.set(0, {
      label: 'home sales',
      dataUrl: 'home-sales.json'
    });

    dataSets.set(1, {
      label: 'gross domestic product',
      dataUrl: 'gross-domestic-product.json'
    });

    dataSets.set(2, {
      label: 'avg. household income',
      dataUrl: 'avg-household-income.json'
    });

    return dataSets;
  }
}
