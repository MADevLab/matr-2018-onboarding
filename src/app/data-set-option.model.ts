export interface DataSetOption {
  label: string;
  dataUrl: string;
}
