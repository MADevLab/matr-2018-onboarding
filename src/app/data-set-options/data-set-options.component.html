<mat-form-field>
  <mat-select [formControl]="dataSetControl"
              placeholder="Select a data set"
              (selectionChange)="changeSet.emit($event.value)">
    <mat-option *ngFor="let o of dataSetOptions"
                [value]="o.value">{{o.label}}</mat-option>
  </mat-select>
</mat-form-field>
<mat-form-field>
  <mat-select [formControl]="dateControl"
              placeholder="View data set at..."
              (selectionChange)="changeDate.emit($event.value)">
    <mat-option *ngFor="let o of dateOptions"
                [value]="o.value">{{o.label}}</mat-option>
  </mat-select>
</mat-form-field>
