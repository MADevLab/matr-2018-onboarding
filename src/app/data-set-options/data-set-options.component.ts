import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { DataSetOption } from '../data-set-option.model';
import { DataPoints, DataSet } from '../data-set.model';

@Component({
  selector: 'app-data-set-options',
  templateUrl: './data-set-options.component.html',
  styleUrls: ['./data-set-options.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DataSetOptionsComponent implements OnChanges {
  @Input() public dataSets: Map<number, DataSetOption>;

  @Output() public changeSet = new EventEmitter<DataSet>();

  @Input() public chosenSet: DataSet;

  @Output() public changeDate = new EventEmitter<number>();

  public dataSetControl = new FormControl({value: -1}, Validators.min(0));

  public dateControl = new FormControl({value: -1, disabled: true}, Validators.min(0));

  public dataSetOptions: OptionEntry[];

  public dateOptions: OptionEntry[];

  public ngOnChanges(changes: SimpleChanges) {
    this.dateOptions = [];

    if (changes.dataSets) {
      this.onDataSetsChange();
    }

    if (changes.chosenSet && this.chosenSet) {
      this.onDataSetSelected();
    }
  }

  public onDataSetsChange() {
    throw new Error('Missing Logic');
  }

  public onDataSetSelected() {
    this.dateControl.enable();
    this.dateOptions = this.chosenSet.data
      .map(this.mapDataPoint)
      .sort(this.sortDateOptions);

    // make sure the listener knows this date exists for the set chosen
    if (this.dateControl.value !== -1 && this.dateOptions.findIndex(date => date.value === this.dateControl.value) !== -1) {
      throw new Error('Missing Logic');
    } else {
      // or let them know to hold 'em
      throw new Error('Missing Logic');
    }
  }

  private mapDataPoint = (dataPoint: DataPoints) => {
    return {
      value: dataPoint.period,
      label: dataPoint.date
    };
  }

  private sortDateOptions = (a: OptionEntry, b: OptionEntry) => {
    if (a.value > b.value) {
      return 1;
    } else if (a.value < b.value) {
      return -1;
    }

    return 0;
  }
}

interface OptionEntry {
  value: number;
  label: string;
}
