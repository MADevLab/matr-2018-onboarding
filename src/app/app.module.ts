import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { FormsModule } from '@angular/forms';

import { MatCardModule } from '@angular/material/card';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { MatIconModule } from '@angular/material/icon';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { DataSetOptionsModule } from './data-set-options/data-set-options.module';
import { MapViewModule } from './map-view/map-view.module';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    HttpClientModule,
    FormsModule,
    MatToolbarModule,
    MatCardModule,
    BrowserAnimationsModule,
    DataSetOptionsModule,
    MapViewModule,
    MatProgressBarModule,
    MatIconModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {}
