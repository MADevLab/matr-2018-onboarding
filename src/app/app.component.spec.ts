import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { DomSanitizer } from '@angular/platform-browser';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { MatCardModule } from '@angular/material/card';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatProgressBarModule, MatProgressBar } from '@angular/material/progress-bar';
import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
import { of, Observable, timer } from 'rxjs';
import { By } from '@angular/platform-browser';

import { DataSetOptionsModule } from './data-set-options/data-set-options.module';
import { MapViewModule } from './map-view/map-view.module';

import { AppComponent } from './app.component';
import { DataSeriesService } from './data-series.service';
import { DataSetOptionsComponent } from './data-set-options/data-set-options.component';
import { DataSetOption } from './data-set-option.model';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let regionsMock: jest.Mock<any>;
  let dataSetMock: jest.Mock<any>;
  let dataSetsSpy: jest.SpyInstance;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        HttpClientModule,
        NoopAnimationsModule,
        MatToolbarModule,
        MatCardModule,
        MatProgressBarModule,
        MatIconModule,
        DataSetOptionsModule,
        MapViewModule
      ],
      declarations: [AppComponent]
    });
    TestBed.compileComponents();

    const sanitizer = TestBed.get(DomSanitizer) as DomSanitizer;
    const iconRegistry = TestBed.get(MatIconRegistry) as MatIconRegistry;

    // mock icon to ignore warnings about not finding a map icon
    iconRegistry
      .addSvgIconLiteral('map', sanitizer.bypassSecurityTrustHtml('<svg><path id="map" name="map"></path</svg>'));

    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;

    const service = TestBed.get(DataSeriesService) as DataSeriesService;

    regionsMock = jest.spyOn(service, 'getRegions').mockReturnValue(of(_regions));
    dataSetMock = jest.spyOn(service, 'getDataSet').mockReturnValue(of(_dataSet));
    dataSetsSpy = jest.spyOn(service, 'getDataSets');
  });

  it('is defined', () => {
    expect(component).toBeDefined();
  });

  it('allows changing the selected region', () => {
    component.changeRegions(_regions);

    expect(component.selectedRegions).toBe(_regions);
  });

  describe('on init', () => {
    it('gets the regions data', () => {
      component.ngOnInit();

      expect(regionsMock).toHaveBeenCalled();
    });

    it('gets the dataSets', async () => {
      component.ngOnInit();

      await timer(3010);

      expect(dataSetsSpy).toHaveBeenCalled();
    });
  });

  describe('when selecting the dataset', () => {
    let _subscription;

    beforeEach(async () => {
      component.ngOnInit();

      // observable will not fire, unless something is subscribing
      _subscription = component.chosenDataSet$.subscribe(o => {});

      component.changeDataSet(1);
    });

    afterEach(() => {
      _subscription.unsubscribe();
    });

    it('sets the concept', () => {
      expect(component.concept).toBeDefined();
      expect(component.concept).toMatchSnapshot();
    });

    it('sets the title', () => {
      expect(component.title).toBeDefined();
      expect(component.title).toMatchSnapshot();
    });

    it('gets the dataset data', () => {
      expect(dataSetMock).toHaveBeenCalled();
    });
  });

  describe('when changing the view date', () => {
    let _subscription;

    beforeEach(async () => {
      component.ngOnInit();

      await timer(3010);

      // observable will not fire, unless something is subscribing
      _subscription = component.chosenDataSet$.subscribe(o => {});

      component.changeDataSet(1);
    });

    afterEach(() => {
      _subscription.unsubscribe();
    });

    it('throws an error if the date is out of range', () => {
      let err = false;

      try {
        component.changeViewDate(700);
      } catch (e) {
        err = true;
      }

      expect(err).toBeTruthy();
    });

    it('returns the data for the date', () => {
      component.changeViewDate(662);

      expect(component.mapData).toMatchSnapshot();
    });
  });

  describe('the ui, before loading the datasets', () => {
    beforeEach(async () => {
      fixture.detectChanges();
      await fixture.whenRenderingDone();
    });

    it('displays a progress bar', () => {
      const el = fixture.debugElement.query(By.css('mat-progress-bar'));

      expect(el).toBeTruthy();
    });

    it('does not display data set options', () => {
      const el = fixture.debugElement.query(By.css('app-data-set-options'));

      expect(el).toBeFalsy();
    });

    it('does not display regions options', () => {
      const el = fixture.debugElement.query(By.css('app-regions-options'));

      expect(el).toBeFalsy();
    });

    it('does not display a map view', () => {
      const el = fixture.debugElement.query(By.css('app-map-view'));

      expect(el).toBeFalsy();
    });
  });

  describe('the ui, after loading the datasets', () => {
    beforeEach(async () => {
      component.ngOnInit = jest.fn();

      const dataSets = new Map<number, DataSetOption>();
      dataSets.set(0, {
        label: 'home sales',
        dataUrl: 'home-sales.json'
      });
      component.dataSets$ = of(dataSets);

      fixture.detectChanges();
      await fixture.whenRenderingDone();
      await fixture.whenStable();
    });

    it('does not display a progress bar', () => {
      const el = fixture.debugElement.query(By.css('mat-progress-bar'));

      expect(el).toBeFalsy();
    });

    it('displays data set options', () => {
      const el = fixture.debugElement.query(By.css('app-data-set-options'));

      expect(el).toBeTruthy();
    });

    it('displays regions options', () => {
      const el = fixture.debugElement.query(By.css('app-regions-options'));

      expect(el).toBeTruthy();
    });

    it('does not a map view, if map data has not been loaded', () => {
      const el = fixture.debugElement.query(By.css('app-map-view'));

      expect(el).toBeFalsy();
    });

    it('displays a map view, if map data has been loaded', async () => {
      component.mapData = _mapData;
      fixture.detectChanges();
      await fixture.whenRenderingDone();
      await fixture.whenStable();

      const el = fixture.debugElement.query(By.css('app-map-view'));

      expect(el).toBeTruthy();
    });
  });
});

const _regions = [
  {
    id: 'NE',
    title: 'Northeast',
    regions: ['CT', 'DC', 'DE']
  },
  {
    id: 'MW',
    title: 'Midwest',
    regions: ['WI']
  },
  {
    id: 'SO',
    title: 'South',
    regions: ['AL', 'AR']
  },
  {
    id: 'WE',
    title: 'West',
    regions: ['WA', 'WY']
  }
];

const _dataSet = {
  concept: 'FGDP$',
  description: 'Gross State Product: Total, (Bil. Chained 2009 $, SAAR)',
  source: 'U.S. Bureau of Economic Analysis (BEA); Moody\'s Analytics Estimated and Forecasted',
  freq: 162,
  transformation: 'None',
  data: [
    {
      period: 661,
      date: '2015Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 662,
      date: '2015Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 664,
      date: '2015Q4',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 663,
      date: '2015Q3',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 665,
      date: '2016Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 666,
      date: '2016Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    }
  ]
};

const _mapData = {
  period: 661,
  date: '2015Q1',
  mapData: [
    { value: 18.1, code: 'ak' },
    { value: 84.58, code: 'al' },
    { value: 44.47, code: 'ar' },
    { value: 112.52, code: 'az' },
    { value: 375.38, code: 'ca' },
    { value: 101.97, code: 'co' },
    { value: 35.69, code: 'ct' },
    { value: 6.36, code: 'dc' },
    { value: 9.6, code: 'de' },
    { value: 435.75, code: 'fl' },
    { value: 142.13, code: 'ga' },
    { value: 16.38, code: 'hi' },
    { value: 53.2, code: 'ia' },
    { value: 36.03, code: 'id' },
    { value: 185.43, code: 'il' },
    { value: 118.85, code: 'in' },
    { value: 38.62, code: 'ks' },
    { value: 62.43, code: 'ky' },
    { value: 55.09, code: 'la' },
    { value: 71.14, code: 'ma' },
    { value: 70.91, code: 'md' },
    { value: 16.34, code: 'me' },
    { value: 156.56, code: 'mi' },
    { value: 86, code: 'mn' },
    { value: 97.64, code: 'mo' },
    { value: 39.08, code: 'ms' },
    { value: 16.16, code: 'mt' },
    { value: 134.4, code: 'nc' },
    { value: 12.5, 'code': 'nd' },
    { value: 33.42, code: 'ne' },
    { value: 18.34, code: 'nh' },
    { value: 89.13, code: 'nj' },
    { value: 24.98, code: 'nm' },
    { value: 65.87, code: 'nv' },
    { value: 143.63, code: 'ny' },
    { value: 223.17, code: 'oh' },
    { value: 62.36, code: 'ok' },
    { value: 62.43, code: 'or' },
    { value: 141.63, code: 'pa' },
    { value: 11.17, code: 'ri' },
    { value: 77.15, code: 'sc' },
    { value: 5.16, code: 'sd' },
    { value: 110.86, code: 'tn' },
    { value: 399.95, code: 'tx' },
    { value: 54.35, code: 'ut' },
    { value: 106.38, code: 'va' },
    { value: 9.6, code: 'vt' },
    { value: 103.66, code: 'wa' },
    { value: 86.11, code: 'wi' },
    { value: 25.16, code: 'wv' },
    { value: 8.84, code: 'wy' }
  ]
};
