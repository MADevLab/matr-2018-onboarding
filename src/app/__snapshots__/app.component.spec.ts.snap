// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent when changing the view date returns the data for the date 1`] = `
Object {
  "date": "2015Q2",
  "mapData": Array [
    Object {
      "code": "ak",
      "value": 49.29,
    },
    Object {
      "code": "al",
      "value": 177.56,
    },
    Object {
      "code": "ar",
      "value": 107.06,
    },
    Object {
      "code": "az",
      "value": 263.13,
    },
    Object {
      "code": "ca",
      "value": 2226.61,
    },
    Object {
      "code": "co",
      "value": 286.63,
    },
    Object {
      "code": "ct",
      "value": 225.09,
    },
  ],
  "period": 662,
}
`;

exports[`AppComponent when selecting the dataset sets the concept 1`] = `"FGDP$"`;

exports[`AppComponent when selecting the dataset sets the title 1`] = `"Gross State Product: Total, (Bil. Chained 2009 $, SAAR)"`;
